src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -246,4 +248,36 @@ public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([F
         var responses = secrets.Select(s => new BaseSecretResponseModel(s));
         return new ListResponseModel<BaseSecretResponseModel>(responses);
     }
+
+    [HttpPost("secrets/{organizationId}/move")]
+    public async Task BulkMoveToProjectAsync(Guid organizationId, [FromBody] BulkMoveToProjectsRequestModel request)
+    {
+        var secrets = (await _secretRepository.GetManyByIds(request.Secrets)).ToList();
+        if (!secrets.Any() || secrets.Count != request.Secrets.Count)
+        {
+            throw new NotFoundException();
+        }
+
+        if (secrets.Any(secret => secret.OrganizationId != organizationId) ||
+            !_currentContext.AccessSecretsManager(organizationId))
+        {
+            throw new NotFoundException();
+        }
+
+        foreach (var secret in secrets)
+        {
+            var authorizationResult = await _authorizationService.AuthorizeAsync(User, secret, SecretOperations.Update);
+            if (!authorizationResult.Succeeded)
+            {
+                throw new NotFoundException();
+            }
+        }
+
+        if (request.Projects.Count != 1)
+        {
+            throw new BadRequestException("You are only allowed to assign a single project.");
+        }
```

Comment:
What are your thoughts on just changing the request model to enforce this restriction? 

```
public class BulkMoveToProjectsRequestModel
{
    [Required]
    public IReadOnlyList<Guid> Secrets { get; set; }

    [Required]
    public Guid Project { get; set; }
}
```

I think it makes the intention clearer from an API contract standpoint.